src/Compiler/Microsoft.AspNetCore.Razor.Language/test/TestFiles/IntegrationTests/ComponentRuntimeCodeGenerationTest/AtTransitions/TestComponent.codegen.cs

```diff
@@ -53,6 +53,14 @@ protected override void BuildRenderTree(global::Microsoft.AspNetCore.Components.
 #line hidden
 #nullable disable
             );
+#nullable restore
+#line (3,23)-(4,1) "x:\dir\subdir\Test\TestComponent.cshtml"
```

Comment:
Should we avoid emitting `#line` pragmas for whitespace nodes? (We only need source mappings right?)

Reply:
I'll let @chsienki comment on this, because he recently made it so that [line pragmas and source mappings were 1:1](https://github.com/dotnet/razor/pull/10055/commits/545fe56e81760a0b7a746d4091b8931a365d8556) in runtime. Obviously at that point whitespace wouldn't have been part of it, but I'm not sure which priority wins.

Reply:
My original assumption here was that at some point in the future, we may not _have_ any source mappings: if we're getting the docs from the generator, we might not have the back channel for the mappings. 

I think with the current architecture that's actually not going to be true, but I do think its a good idea to maintain a 1-1 mapping, so the source doc contains all the information (especially for some of our strategic partners who are using the line pragmas only)